Source/SwiftLintFramework/Rules/Lint/UnusedImportRule.swift

```diff
@@ -129,11 +161,36 @@ private extension File {
         }
         // Always disallow 'import Swift' because it's available without importing.
         usrFragments = usrFragments.filter { $0 != "Swift" }
-        let unusedImports = imports.filter { !usrFragments.contains($0) }
+        var unusedImports = imports.filter { !usrFragments.contains($0) }
+        // Certain Swift attributes requires importing Foundation.
+        if unusedImports.contains("Foundation") && containsAttributesRequiringFoundation() {
+            unusedImports.removeAll(where: { $0 == "Foundation" })
+        }
         return unusedImports.map { module in
-            return (module, contents.bridge().range(of: "import \(module)\n"))
+            let testableImportRange = contentsNSString.range(of: "@testable import \(module)\n")
+            if testableImportRange.location != NSNotFound {
+                return (module, testableImportRange)
+            }
+
+            return (module, contentsNSString.range(of: "import \(module)\n"))
         }
     }
+
+    private func containsAttributesRequiringFoundation() -> Bool {
+        guard contents.contains("@objc") else {
```

Comment:
Do `IBAction` and `IBOutlet` also require this logic? I assume so but maybe it's less likely to have a file with one of those where it doesn't also require the import for other reasons.

Reply:
My thinking exactly. Let's cross that bridge when we get there.